### PR TITLE
Set smp to vcpu_maxcpus

### PR DESCRIPTION
**Pull request by qingwangrh**
Fix bug cpu topology: value of sockets * dies * cores * threads less than smp_cpus
ID:1793766
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=virtio_scsi_mq.default.q35  --guestname=RHEL.8.2.0 --imageformat=qcow2 --driveformat=virtio_scsi --nicmodel=virtio_net --machines=q35 --platform=x86_64 --clone=no

 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_scsi_mq.q35: PASS (103.70 s)

**Comment by qingwangrh** [comment]
@YongxueHong @peixiu @zhenyzha Please help to review.

**Comment by peixiu** [comment]
> python ConfigTest.py --testcase=virtio_scsi_mq.default.q35 --guestname=RHEL.8.2.0 --imageformat=qcow2 --driveformat=virtio_scsi --nicmodel=virtio_net --machines=q35 --platform=x86_64 --clone=no
> 
> (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_scsi_mq.q35: PASS (103.70 s)

please test for a windows guest as well, thanks~

File: qemu/tests/virtio_scsi_mq.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def wmi_facility_test(session):
 
     cpu_irq_map = {}
     timeout = float(params.get("login_timeout", 240))
-    num_queues = str(utils_cpu.online_cpus_count())
+    num_queues = str(params['vcpu_maxcpus'])
```
Comment (zhenyzha) [inline qemu/tests/virtio_scsi_mq.py]: no need for str (), params['vcpu_maxcpus']  type is str.

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=virtio_scsi_mq --guestname=Win2019 --imageformat=qcow2 --driveformat=virtio_scsi --nicmodel=virtio_net --machines=q35 --platform=x86_64 --clone=no
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_scsi_mq.wmi_facility_check.q35: PASS (65.38 s)

File: qemu/tests/virtio_scsi_mq.py (hunk 1 above)
Comment (qingwangrh) [reply qemu/tests/virtio_scsi_mq.py]: updated

**Comment by peixiu** [comment]
I tested on win10-32 guest os, it's failed as follows, I see you closed this bug https://bugzilla.redhat.com/show_bug.cgi?id=1774815 as duplicate, the error is same with this bug comment#0, could you help to fix it as well? thanks~

python3 ConfigTest.py --testcase=virtio_scsi_mq --guestname=Win10 --platform=i386 --driveformat=virtio_scsi --clone=no --rng_type=no_virtio_rng --machines=q35

 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_scsi_mq.wmi_facility_check.q35: FAIL: The queue_num from guest is not match with expected.\nqueue_num from guest is [], expected is 24 (135.89 s)

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.virtio_scsi_mq: PASS (1470.95 s)

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Comment by peixiu** [comment]
PASS 1-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_scsi_mq.wmi_facility_check.q35

Hit product bug before, tested with latest virtio-win-prewhql-180 version, it can be passed with smp<=32. And reopen this issue https://bugzilla.redhat.com/show_bug.cgi?id=1774815 to track smp>32 issue.

Acked-By: Peixiu Hou phou@redhat.com

**Pull request merged**